### PR TITLE
Media & Text: Allow drag n drop media replacement

### LINKED ISSUE #26320: I should be able to drag and drop to replace media uploaded to the Media & Text block
**Is your feature request related to a problem? Please describe.**
While we allow drag and drop functionality to image and gallery blocks, the Media & Text block does not allow it. 

**Screenshot**
github

**Describe the solution you'd like**
A dropzone component that exists within the Media & Text block (only covering the media, not the text) where I can drag and drop an image, or video, to replace it. 

**Describe alternatives you've considered**

### PR DESCRIPTION
**Pull request by Mamaduka**
## Description
Adds dropzone to the media part of the block when an image/video is already set. Fixes #26320.

## How has this been tested?
Manually on local dev environment.

Steps:
1. Add the "Media & Text" block to the post.
2. Add media.
3. Drag and drop media replacement.
4. See if the dropzone indicator appears.
5. See if media is replaced.

## Types of changes
New feature

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by draganescu** [comment]
Howdy @Mamaduka and welcome! Thank you for contributing. I tested this PR and it does work, but it would be great if the block would show something is happening upon drag. Here is a video comparing to how the image block handles drag and drop:


https://user-images.githubusercontent.com/107534/110692620-3289f180-81ef-11eb-93f7-9385199f9283.mp4

**Comment by Mamaduka** [comment]
Thanks, @draganescu for testing the PR.

100% with you about the upload indicator.

I'm reusing the `PlaceholderContainer` component. I'm not sure if those changes should be part of this PR or be a separate one.

**Comment by Mamaduka** [comment]
Hi, @draganescu 

After thinking more about this, it makes sense to include Spinner change as a part of this PR. See b4033b3.

**Comment by Mamaduka** [comment]
Hi, @draganescu

Is it possible to get another review on this PR? 🙇

**Comment by jasmussen** [comment]
Sorry about the delay on this one. In fact the branch was so old I had to rebase it in order to compile it 🙈 

You can always feel free to request a review from me in the future if you get stalled.

Here's before:

before

Here's after:

after

Very nice. It could use a code sanity check, and I'd recommend giving it a good rebase first. But giving it a green check for the experience. Excellent work.

**Review by jasmussen — APPROVED** [review]

**Comment by Mamaduka** [comment]
Thanks for the review, @jasmussen.

Rebased and will merge once all checks are green 🤞

**Comment by draganescu** [comment]
Thanks for chiming in @jasmussen and sorry for not following up @Mamaduka! Nevertheless 🎉 great work!

**Pull request merged**
